### PR TITLE
Remove astropy helpers

### PR DESCRIPTION
**Pull request by mwcraig**
This is intended solely to move away from helpers. I haven't done much with the travis matrix yet; I just made the minimal changes needed to make the tests run (maybe)

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +0,0 @@
-[submodule "astropy_helpers"]
```
Comment (pllim) [inline .gitmodules]: Why not just remove the entire file? In your branch here, I see an empty `.gitmodules` file.

File: .travis.yml [hunk 2]
```diff
@@ -128,9 +124,11 @@ install:
...
+   - pip install -e .
    - $MAIN_CMD $SETUP_CMD
 
 after_success:
-    # If coveralls.io is set up for this package, uncomment the line below.
-    # The coveragerc file may be customized as needed for your package.
-    # - if [[ $SETUP_CMD == *coverage* ]]; then coveralls --rcfile='astrowidgets/tests/coveragerc'; fi
+    # Upload coverage to codecov
+    - pip install codecov
+    - codecov
```
Comment (pllim) [inline .travis.yml]: Note: Need to double check that it uploads properly for the run after the merge of this PR.

File: .travis.yml [hunk 3]
```diff
@@ -24,8 +24,8 @@ env:
         - NUMPY_VERSION=stable
         - ASTROPY_VERSION=stable
         - GINGA_VERSION=stable
-        - MAIN_CMD='python setup.py'
-        - SETUP_CMD='test'
+        - MAIN_CMD='pytest'
```
Comment (pllim) [inline .travis.yml]: Why not use `tox` here too?

File: .gitignore [hunk 4]
```diff
@@ -10,13 +10,13 @@ __pycache__
...
 
 # Other generated files
-*/version.py
 */cython_version.py
 htmlcov
 .coverage
 MANIFEST
 .ipynb_checkpoints
 example_notebooks/test.png
+*/version.py
```
Comment (pllim) [inline .gitignore]: I found myself also have to ignore `pip-wheel-metadata/` for my packages. 🤷‍♀ 
```suggestion
*/version.py
pip-wheel-metadata/
```

File: astrowidgets/_astropy_init.py [hunk 5]
```diff
@@ -17,11 +17,6 @@
     from .version import version as __version__
 except ImportError:
     __version__ = ''
-try:
-    from .version import githash as __githash__
-except ImportError:
-    __githash__ = ''
-
 
 if not _ASTROPY_SETUP_:  # noqa
```
Comment (pllim) [inline astrowidgets/_astropy_init.py]: Don't need `_ASTROPY_SETUP_` no more throughout this file. As a result, you can also dedent the config stuff.

File: astrowidgets/tests/setup_package.py [hunk 6]
```diff
@@ -1,11 +0,0 @@
-# import os
-
-# If this package has tests data in the tests/data directory, add them to
-# the paths here, see commented example
-paths = ['coveragerc',
```
Comment (pllim) [inline astrowidgets/tests/setup_package.py]: This is still package data and needs to be moved to `setup.cfg`.

File: setup.cfg [hunk 7]
```diff
@@ -31,20 +35,35 @@ exclude = setup_package.py,conftest.py,__init__.py
...
-install_requires = astropy, ginga, pillow, ipywidgets>=7.5, ipyevents>=0.6.3, jupyterlab>=1, aggdraw
-# version should be PEP386 compatible (http://www.python.org/dev/peps/pep-0386)
-version = 0.2.0.dev
-# Note: you will also need to change this in your package's __init__.py
-minimum_python_version = 3.6
+
+[options]
+zip_safe = False
+packages = find:
+setup_requires = setuptools_scm
```
Comment (pllim) [inline setup.cfg]: Not needed.
```suggestion
```

File: setup.py [hunk 8]
```diff
@@ -1,141 +1,51 @@
...
-      url=URL,
-      long_description=LONG_DESCRIPTION,
-      cmdclass=cmdclassd,
-      zip_safe=False,
-      use_2to3=False,
-      entry_points=entry_points,
-      python_requires='>={}'.format(__minimum_python_version__),
-      **package_info
-)
+setup(use_scm_version={'write_to': Path('astrowidgets') / 'version.py'})
```
Comment (pllim) [inline setup.py]: Oh, is `Path` necessary for Windows? I didn't use it.

**Comment by pllim** [comment]
Also, directly on this repo, there is a need to tag a dev version (`0.1.2.dev`?) for `setuptools_scm` to work properly.

File: setup.py (hunk 8 above)
Comment (mwcraig) [reply setup.py]: I had assumed it was, but don't know for sure.

File: .gitmodules (hunk 1 above)
Comment (mwcraig) [reply .gitmodules]: Removed, thanks.

File: setup.py (hunk 8 above)
Comment (pllim) [reply setup.py]: Hmm... Windows seems to pick up the version without `Path`, so maybe `setuptools_scm` is smart enough with a plain text slash, but I guess it does not hurt this way either. From my test log without using `Path`:

```
============================= test session starts =============================
platform win32 -- Python 3.7.5, pytest-5.3.2, py-1.8.0, pluggy-0.13.1

Running tests with stsynphot version 0.3.dev11+g9902e04.
```

**Comment by mwcraig** [comment]
> Also, directly on this repo, there is a need to tag a dev version (0.1.2.dev?) for setuptools_scm to work properly.

I think we are fine without the dev tag since this repo just has a single branch. Since it has `0.1.1` tagged on it `setuptools_scm` is automatically generating a the version `0.1.2.dev`.

File: setup.py (hunk 8 above)
Comment (mwcraig) [reply setup.py]: ok, I'll pull it out. Less code is better!

File: .travis.yml [hunk 9]
```diff
@@ -61,12 +61,10 @@ matrix:
     fast_finish: true
 
     include:
-        - stage: Initial tests
-          env: SETUP_CMD='egg_info'
-
         # Do a coverage test.
         - stage: Initial tests
-          env: SETUP_CMD='test --coverage'
+          env: MAIN_CMD='tox -e'
```
Comment (pllim) [inline .travis.yml]: `MAIN_CMD` here is now repetitive.

File: astrowidgets/__init__.py [hunk 10]
```diff
@@ -17,7 +17,3 @@ class UnsupportedPythonError(Exception):
         pass
 
     raise UnsupportedPythonError("astrowidgets does not support Python < {}".format(__minimum_python_version__))
-
-if not _ASTROPY_SETUP_:
-    # For egg_info test builds to pass, put package imports here.
-    from .core import *
```
Comment (pllim) [inline astrowidgets/__init__.py]: You can still do `from .core import *`  here, but you don't need to do it inside the `if` block. 😅

Comment (mwcraig) [reply astrowidgets/__init__.py]: 😮 🤦‍♂ Not sure how the tests passed with this...

**Comment by mwcraig** [comment]
What is `TESTED_VERSIONS`?

**Comment by pllim** [comment]
@mwcraig ,  https://github.com/astropy/pytest-astropy-header explains how to use `TESTED_VERSIONS`. 😸

**Comment by mwcraig** [comment]
OK, assuming the tests pass with the latest config cleanup here, I'll merge then open an issue as a reminder about cleaning up `conftest.py` and `conf.py`

**Comment by pllim** [comment]
Sounds good! Feel free to merge when you think you are done with what you have planned for this PR. Thanks!

**Pull request merged**